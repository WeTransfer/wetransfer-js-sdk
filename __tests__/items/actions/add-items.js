const routes = require('../../../src/config/routes');
const addItemsAction = require('../../../src/items/actions/add-items');
const {
  normalizeItem,
  normalizeResponseItem
} = require('../../../src/items/models');

describe('Add items action', () => {
  let addItems = null;
  const mocks = {};
  beforeEach(() => {
    mocks.request = { send: jest.fn() };
    mocks.request.send.mockReturnValue([
      {
        id: 'random-hash',
        content_identifier: 'file',
        local_identifier: 'delightful-cat',
        meta: {
          multipart_parts: 3,
          multipart_upload_id: 'some.random-id--'
        },
        name: 'kittie.gif',
        size: 195906,
        upload_id: 'more.random-ids--',
        upload_expires_at: 1520410633
      },
      {
        id: 'random-hash',
        content_identifier: 'web_content',
        meta: {
          title: 'WeTransfer'
        },
        url: 'https://wetransfer.com'
      }
    ]);

    addItems = addItemsAction({
      routes,
      request: mocks.request,
      normalizeItem,
      normalizeResponseItem
    });
  });

  it('should create an add items request', async () => {
    const items = await addItems('transfer-id', [
      {
        local_identifier: 'wetransfer.com',
        content_identifier: 'web_content',
        url: 'https://wetransfer.com',
        meta: {
          title: 'WeTransfer'
        }
      }
    ]);
    expect(items).toMatchSnapshot();
  });
});
