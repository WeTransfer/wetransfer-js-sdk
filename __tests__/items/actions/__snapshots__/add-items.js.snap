// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add items action should create an add items request 1`] = `
Array [
  Object {
    "content_identifier": "file",
    "id": "random-hash",
    "local_identifier": "delightful-cat",
    "meta": Object {
      "multipart_parts": 3,
      "multipart_upload_id": "some.random-id--",
    },
    "name": "kittie.gif",
    "size": 195906,
  },
  Object {
    "content_identifier": "web_content",
    "id": "random-hash",
    "meta": Object {
      "title": "WeTransfer",
    },
  },
]
`;
